### PR TITLE
Adds support for fetching external block explorers

### PR DESCRIPTION
**Pull request by douglance**
- Fetches block explorer data from a [Github Pages location](https://gridplus.github.io/chains/chains.json) that is generated from [a GridPlus repo](https://github.com/GridPlus/chains/)
- Bumps version to 2.1.0 because of the updated behavior. 
- Tested using the `'[EVM] Test decoders'` set of tests.

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -528,22 +528,29 @@ function getFwVersionConst (v) {
 // eslint-disable-next-line no-control-regex
 const ASCII_REGEX = /^[\x00-\x7F]+$/;
 
+const EXTERNAL_NETWORKS_BY_CHAIN_ID_URL =
+  'https://gridplus.github.io/chains/chains.json';
```
Comment (alex-miller-0) [inline src/constants.ts:532]: Interesting... how did you map to this URL?

File: src/util.ts [hunk 2]
```diff
@@ -432,10 +457,13 @@ export async function fetchCalldataDecoder (_data: Uint8Array | string, to: stri
...
     const chainId = Number(_chainId);
-    const supportedChain = NETWORKS_BY_CHAIN_ID[chainId];
+    const supportedChainList =
+      (await fetchExternalNetworkByChainId()) ?? NETWORKS_BY_CHAIN_ID;
+    const supportedChain = supportedChainList[chainId];
     if (supportedChain) {
       const baseUrl = supportedChain.baseUrl;
-      const url = `${baseUrl}/api?module=contract&action=getabi&address=${to}&apiKey=${process.env.ETHERSCAN_KEY}`
+      const apiRoute = supportedChain.apiRoute;
+      const url = `${baseUrl}/${apiRoute}&address=${to}`;
```
Comment (alex-miller-0) [inline src/util.ts:466]: You've duplicated `&address=` in all provided `apiRoute` entries and here

**Review by alex-miller-0 — CHANGES_REQUESTED** [review]

File: src/constants.ts (hunk 1 above)
Comment (douglance) [reply src/constants.ts:532]: The `chains` repo deploys whatever is in its `/data` folder to github pages with [this workflow.](https://github.com/GridPlus/chains/blob/main/.github/workflows/build-deploy.yml) So the `chains.json` file is deployed there and we can grab it here.

Comment (douglance) [reply src/constants.ts:532]: Commits to `main` in that repo will automatically deploy an update.

Comment (alex-miller-0) [reply src/constants.ts:532]: 😮

File: src/util.ts [hunk 3]
```diff
@@ -432,10 +457,13 @@ export async function fetchCalldataDecoder (_data: Uint8Array | string, to: stri
     // Convert the chainId to a number and use it to determine if we can call out to
     // an etherscan-like explorer for richer data.
     const chainId = Number(_chainId);
-    const supportedChain = NETWORKS_BY_CHAIN_ID[chainId];
+    const supportedChainList =
```
[the remark this answers is not in the document]
Comment (douglance) [reply src/util.ts]: fixed 7aff393e2a028330b7d94a6041c6458d48e73fa9

File: src/util.ts [hunk 4]
```diff
@@ -432,10 +458,14 @@ export async function fetchCalldataDecoder (_data: Uint8Array | string, to: stri
...
     // an etherscan-like explorer for richer data.
     const chainId = Number(_chainId);
-    const supportedChain = NETWORKS_BY_CHAIN_ID[chainId];
+    const cachedNetwork = NETWORKS_BY_CHAIN_ID[chainId];
+    const supportedChain = cachedNetwork
+      ? cachedNetwork
+      : await fetchExternalNetworkForChainId(chainId);
     if (supportedChain) {
       const baseUrl = supportedChain.baseUrl;
-      const url = `${baseUrl}/api?module=contract&action=getabi&address=${to}&apiKey=${process.env.ETHERSCAN_KEY}`
```
Comment (alex-miller-0) [inline src/util.ts:438]: Oooh we don't want to lose that API key thing or testing will be painfully slow. Should add the `apiKey` arg conditional on `process.env.ETHERSCAN_KEY` existing

Comment (alex-miller-0) [reply src/util.ts:438]: Ehhh not to be too nitpicky but I would prefer you conditionally add the argument rather than pass `apiKey=undefined` to the url

**Review by alex-miller-0 — APPROVED** [review]

**Pull request merged**